### PR TITLE
feat(clean): add `--to` filter for `git gtr clean --merged`

### PR DESCRIPTION
**Pull request by scarf005**
# Pull Request

## Description

Adds `--to <Ref>` option to `git gtr clean --merged` and tightens merged cleanup so only worktrees whose current branch tip matches a merged PR/MR for the requested target ref are considered.

## Motivation

For teams that merge PRs into staging or release branches before `main`, `git gtr clean --merged --to main` must not treat staging-only merges as eligible for deletion. This also needs to avoid false positives when branch names are reused.

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Code refactoring (no functional changes)
- [ ] Other (please describe):

## Testing

- `bats tests/cmd_clean.bats tests/provider.bats tests/cmd_help.bats tests/completion.bats`
- `shellcheck lib/commands/clean.sh lib/provider.sh lib/commands/help.sh`
- `./bin/gtr clean`
- `git gtr clean`
- `./scripts/generate-completions.sh --check`

### Manual Testing Checklist

**Tested on:**

- [x] macOS
- [ ] Linux (specify distro: **__**)
- [ ] Windows (Git Bash)

**Core functionality tested:**

- [x] `git gtr new <branch>` - Create worktree
- [x] `git gtr go <branch>` - Navigate to worktree
- [ ] `git gtr editor <branch>` - Open in editor (if applicable)
- [ ] `git gtr ai <branch>` - Start AI tool (if applicable)
- [ ] `git gtr rm <branch>` - Remove worktree
- [ ] `git gtr list` - List worktrees
- [ ] `git gtr config` - Configuration commands (if applicable)
- [x] Other commands affected by this change: `git gtr clean --merged --to <ref>`

### Test Steps

1. Create a worktree and commit on the feature branch.
2. Open a PR targeting a non-main branch such as `dummy`.
3. Run `git gtr clean --merged --to main`.
4. Verify the worktree is not removed because the PR was not merged into `main`.
5. Reuse a previously merged branch name and verify cleanup only matches the current branch tip.

**Expected behavior:**
Only worktrees whose current branch tip belongs to a merged PR/MR into the specified target ref are removed.

**Actual behavior:**
`git gtr clean --merged --to main` ignores non-merged worktrees, rejects `--to` without `--merged`, and does not emit dirty-skip warnings for branches that were never eligible for deletion.

## Breaking Changes

- [ ] This PR introduces breaking changes
- [ ] I have discussed this in an issue first
- [ ] Migration guide is included in documentation

## Checklist

Before submitting this PR, please check:

- [x] I have read [CONTRIBUTING.md](../CONTRIBUTING.md)
- [x] My code follows the project's style guidelines
- [x] I have performed manual testing on at least one platform
- [x] I have updated documentation (README.md, CLAUDE.md, etc.) if needed
- [ ] My changes work on multiple platforms (or I've noted platform-specific behavior)
- [x] I have added/updated shell completions (if adding new commands or flags)
- [x] I have tested with both `git gtr` (production) and `./bin/gtr` (development)
- [x] No new external dependencies are introduced (Bash + git only)
- [x] All existing functionality still works

## Additional Context

This also fixes the completion generator so regenerated bash/zsh/fish completions keep the `--to` option.

---

## License Acknowledgment

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache License 2.0](../LICENSE.txt).

## Summary by CodeRabbit

* **New Features**
  * Added --to <ref> for git gtr clean --merged to restrict cleanup to worktrees whose PRs/MRs were merged into the specified base ref; --to is only valid with --merged.

* **Documentation**
  * Updated README and command help with --to usage and an example (git gtr clean --merged --to main).

* **Shell Completions**
  * Added --to completion support for Bash, Zsh, and Fish.

* **Tests**
  * Added tests for --to validation, merged-check behavior, and related edge cases.

### CODE REVIEW

File: completions/_git-gtr [hunk 1]
```diff
@@ -80,6 +80,7 @@ _git-gtr() {
   if (( CURRENT >= 4 )) && [[ $words[3] == clean ]]; then
     _arguments \
       '--merged[Remove worktrees with merged PRs/MRs]' \
+      '--to[Only remove worktrees for PRs/MRs merged into this ref]:ref:' \
       '--yes[Skip confirmation prompts]' \
```
[the remark this answers is not in the document]
Comment (helizaga) [reply completions/_git-gtr:85]: Already addressed in current code.

File: lib/provider.sh [hunk 2]
```diff
@@ -97,23 +97,59 @@ ensure_provider_cli() {
...
     github)
-      local pr_state
-      pr_state=$(gh pr list --head "$branch" --state merged --json state --jq '.[0].state' 2>/dev/null || true)
-      [ "$pr_state" = "MERGED" ]
+      local pr_matches
+      if [ -n "$target_ref" ]; then
+        if [ -n "$branch_tip" ]; then
+          pr_matches=$(gh pr list --head "$branch" --base "$target_ref" --state merged --json state,headRefOid --jq "map(select(.state == \"MERGED\" and .headRefOid == \"$branch_tip\")) | length" 2>/dev/null || true)
+        else
+          pr_matches=$(gh pr list --head "$branch" --base "$target_ref" --state merged --json state --jq 'map(select(.state == "MERGED")) | length' 2>/dev/null || true)
```
[the remark this answers is not in the document]
Comment (helizaga) [reply lib/provider.sh]: Fixed in ad23522.

File: lib/provider.sh [hunk 3]
```diff
@@ -97,23 +97,59 @@ ensure_provider_cli() {
...
+        fi
+      else
+        if [ -n "$branch_tip" ]; then
+          pr_matches=$(gh pr list --head "$branch" --state merged --json state,headRefOid --jq "map(select(.state == \"MERGED\" and .headRefOid == \"$branch_tip\")) | length" 2>/dev/null || true)
+        else
+          pr_matches=$(gh pr list --head "$branch" --state merged --json state --jq 'map(select(.state == "MERGED")) | length' 2>/dev/null || true)
```
[the remark this answers is not in the document]
Comment (helizaga) [reply lib/provider.sh]: Fixed in ad23522.

**Comment by scarf005** [comment]
@helizaga hi, would there be anything left for me to fix? thanks!
